### PR TITLE
fix(ci): Fix CI issues

### PR DESCRIPTION
**Pull request by roninjin10**
Update ci builder and also do a temporary fix to fix ci while ci builder is redeployed

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6287** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6287" target="_blank">Graphite</a>  👈
    * **PR #6210** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6210" target="_blank">Graphite</a> 
      * **PR #6303** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6303" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6287?utm_source=stack-comment).

**Review by Inphi — APPROVED** [review]
LGTM! Thanks

File: .circleci/config.yml [hunk 1]
```diff
@@ -388,6 +409,13 @@ jobs:
           name: Install pnpm package manager
           command: |
             npm i pnpm --global
+      # A github dep clones-with-immutable-args is installed via github
+      # packages installed via npm via github automatically run postpack scripts
+      # their postpack script happens to use yarn so we need it here 
+      - run:
+          name: Install yarn package manager
```
Comment (mslipper) [inline .circleci/config.yml:424]: Why not install `yarn` and `pnpm` inside the `ci-builder`? CI builder is supposed to contain all of our build dependencies.

**Review by mslipper — CHANGES_REQUESTED** [review]
See comment re: `ci-builder`.

File: .circleci/config.yml [hunk 2]
```diff
@@ -107,6 +107,13 @@ jobs:
...
+          command: |
+            npm i yarn@1 --global
```
Comment (roninjin10) [inline .circleci/config.yml:118]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 3]
```diff
@@ -327,6 +334,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:347]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 4]
```diff
@@ -361,6 +375,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:390]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml (hunk 1 above)
Comment (roninjin10) [reply .circleci/config.yml:424]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 5]
```diff
@@ -464,6 +492,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:511]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 6]
```diff
@@ -489,6 +524,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:545]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 7]
```diff
@@ -506,6 +548,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:571]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 8]
```diff
@@ -541,6 +590,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:615]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 9]
```diff
@@ -564,6 +620,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:647]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 10]
```diff
@@ -597,6 +660,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:689]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 11]
```diff
@@ -670,6 +740,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:771]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml [hunk 12]
```diff
@@ -746,6 +823,13 @@ jobs:
...
```
Comment (roninjin10) [inline .circleci/config.yml:857]: this yarn install and the pnpm install here and above can both be removed in a future pr after ci builder publishes

File: .circleci/config.yml (hunk 2 above)
Comment (refcell) [reply .circleci/config.yml:118]: Can we stack those changes ontop this pr as a draft so we don't forget?

**Review by tynes — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Pull request merged**